### PR TITLE
Implements start container.

### PR DESCRIPTION
**Pull request by feiskyer**
This PR makes a workaround to start container by restarting the pod.

```sh
$ cluster/kubectl.sh run nginx --image=nginx
deployment "nginx" created

$ cluster/kubectl.sh describe pods
Name:		nginx-701339712-5helf
Namespace:	default
Node:		127.0.0.1/127.0.0.1
Start Time:	Thu, 17 Nov 2016 06:06:56 +0000
Labels:		pod-template-hash=701339712
		run=nginx
Status:		Running
IP:		192.168.140.50
Controllers:	ReplicaSet/nginx-701339712
Containers:
  nginx:
    Container ID:	hyper://e1e5988b92d87e1c878f35c81d744d0503b1e521cd61484a3fa203db6754e067
    Image:		nginx
    Image ID:		sha256:05a60462f8bafb215ddc5c20a364b5fb637670200a74a5bb13a1b23f64515561
    Port:
    State:		Running
      Started:		Thu, 17 Nov 2016 06:07:06 +0000
    Ready:		True
    Restart Count:	0
    Volume Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-w1ve3 (ro)
    Environment Variables:	<none>
Conditions:
  Type		Status
  Initialized 	True
  Ready 	True
  PodScheduled 	True
Volumes:
  default-token-w1ve3:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-w1ve3
QoS Class:	BestEffort
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath		Type		Reason		Message
  ---------	--------	-----	----			-------------		--------	------		-------
  53s		53s		1	{default-scheduler }				Normal		Scheduled	Successfully assigned nginx-701339712-5helf to 127.0.0.1
  53s		53s		1	{kubelet 127.0.0.1}				Normal		SandboxReceived	Pod sandbox received, it will be created.
  48s		48s		1	{kubelet 127.0.0.1}	spec.containers{nginx}	Normal		Pulling		pulling image "nginx"
  47s		47s		1	{kubelet 127.0.0.1}	spec.containers{nginx}	Normal		Pulled		Successfully pulled image "nginx"
  47s		47s		1	{kubelet 127.0.0.1}	spec.containers{nginx}	Normal		Created		Created container with id e1e5988b92d87e1c878f35c81d744d0503b1e521cd61484a3fa203db6754e067
  43s		43s		1	{kubelet 127.0.0.1}	spec.containers{nginx}	Normal		Started		Started container with id e1e5988b92d87e1c878f35c81d744d0503b1e521cd61484a3fa203db6754e067
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/42)

### CODE REVIEW

File: pkg/hyper/container.go [hunk 1]
```diff
@@ -18,11 +18,32 @@ package hyper
...
 	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
 
 	"k8s.io/frakti/pkg/hyper/types"
 	kubeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 )
 
+// CreateContainer creates a new container in specified PodSandbox
```
Comment (feiskyer) [inline pkg/hyper/container.go]: Agree, that's more clear.

File: pkg/hyper/container.go [hunk 2]
```diff
@@ -64,3 +85,203 @@ func buildUserContainer(config *kubeapi.ContainerConfig, sandboxConfig *kubeapi.
...
+		ImageRef:    &status.Container.ImageID,
+		Metadata:    containerMetadata,
+		State:       &state,
+		Labels:      kubeletLabels,
+		Annotations: annotations,
+		CreatedAt:   &createdAtNano,
+	}
+
+	mounts := make([]*kubeapi.Mount, len(status.Container.VolumeMounts))
+	for idx, mnt := range status.Container.VolumeMounts {
```
Comment (resouer) [inline pkg/hyper/container.go]: Seems we do not support volumes mounts, but I guess this part does not break anything?

File: pkg/hyper/container.go [hunk 3]
```diff
@@ -41,16 +62,16 @@ func buildUserContainer(config *kubeapi.ContainerConfig, sandboxConfig *kubeapi.
...
+	// TODO: support volume mounts in upstream hyperd.
+	// volumes := make([]*types.UserVolumeReference, len(config.Mounts))
+	// for idx, v := range config.Mounts {
+	// 	volumes[idx] = &types.UserVolumeReference{
+	// 		Volume:   v.GetHostPath(),
+	// 		Path:     v.GetContainerPath(),
+	// 		ReadOnly: v.GetReadonly(),
+	// 	}
+	// }
+	// containerSpec.Volumes = volumes
```
Comment (resouer) [inline pkg/hyper/container.go]: May be better to leave an empty volumes list instead of `nil`?

File: pkg/hyper/container.go (hunk 2 above)
Comment (feiskyer) [inline pkg/hyper/container.go]: yes, it doesn't break anything else. it is only for ContainerStatus.

File: pkg/hyper/container.go (hunk 3 above)
Comment (feiskyer) [inline pkg/hyper/container.go]: ack.

File: pkg/hyper/container.go [hunk 4]
```diff
@@ -64,3 +85,203 @@ func buildUserContainer(config *kubeapi.ContainerConfig, sandboxConfig *kubeapi.
...
+	// pod to workaround this.
+	// TODO: replace this with real StartContainer after hyperd's refactoring.
+	container, err := h.client.GetContainerInfo(rawContainerID)
+	if err != nil {
+		glog.Errorf("Failed to get container %q: %v", rawContainerID, err)
+		return err
+	}
+	_, reason, err := h.client.StopPod(container.PodID)
+	if err != nil {
+		glog.Errorf("Failed to stop pod %q with reason (%q): %v", container.PodID, reason, err)
```
Comment (feiskyer) [inline pkg/hyper/container.go]: ack.

**Pull request merged**